osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/AdHocManagementServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package org.opensmartgridplatform.adapter.domain.core.application.services;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.opensmartgridplatform.adapter.domain.core.infra.jms.core.OsgpCoreRequestMessageSender;
+import org.opensmartgridplatform.domain.core.entities.Device;
+import org.opensmartgridplatform.domain.core.entities.Organisation;
+import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
+import org.opensmartgridplatform.domain.core.services.DeviceDomainService;
+import org.opensmartgridplatform.domain.core.services.OrganisationDomainService;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.infra.jms.RequestMessage;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class AdHocManagementServiceTest{
+
+    @Mock
+    private DeviceDomainService deviceDomainService;
+    
+    @Mock
+    private OsgpCoreRequestMessageSender osgpCoreRequestMessageSender;
+
+    @Mock
+    private Device device;
+
+    @Mock
+    private OrganisationDomainService organisationDomainService;
+
+    
+    @InjectMocks
+    private AdHocManagementService adHocManagementService;
+    
+    @Test
+    public void testSetReboot() throws FunctionalException, UnknownEntityException {
+		when(this.device.getIpAddress()).thenReturn("127.0.0.1");
+		when(this.organisationDomainService.searchOrganisation(any(String.class))).thenReturn(Mockito.mock(Organisation.class));
```

Comment:
I don't think setting up the `organisationDomainService` mock this way gives the result you would want to see in the test.
If you remove this line the outcome does not change, which is a clear indication something is wrong.
This is in part caused by the effect that the code from `setReboot` does not care about the actual return value of `searchOrganisation`. It does however care whether an exception is thrown or not.

I think that you could get the verification that you would like to see, by adding the following method and calling it from the test with the organisation identification that is expected:

`void givenOnlyTheIdentifiedOrganisationExists(final String organisationIdentification) throws Exception {`
`    when(this.organisationDomainService.searchOrganisation(anyString())).thenAnswer(new Answer<Organisation>() {`
`        @Override`
`        public Organisation answer(final InvocationOnMock invocation) throws Throwable {`
`            final String actualIdentification = invocation.getArgument(0);`
`            if (organisationIdentification.equals(actualIdentification)) {`
`                return Mockito.mock(Organisation.class);`
`            }`
`            throw new UnknownEntityException(Organisation.class, actualIdentification);`
`        }`
`    });`
`}`